odoo/api.py

```diff
@@ -34,12 +33,31 @@
 from .tools.translate import _
 from odoo.tools.misc import StackMap
 
-if TYPE_CHECKING:
+import typing
+if typing.TYPE_CHECKING:
+    from collections.abc import Callable
     from odoo.sql_db import Cursor, TestCursor
     from odoo.models import BaseModel
+    from .models import NewId
+
+    try:
+        from typing_extensions import Self  # noqa: F401
+    except ImportError:
+        from typing import Self  # noqa: F401
+    M = typing.TypeVar("BaseModel", bound=BaseModel)
```

Comment:
Variable name and first argument must be aligned. (same in the else block)
```suggestion
    M = typing.TypeVar("M", bound=BaseModel)
```